File: backend/internal/middleware/cors.go
package middleware

import (
	"github.com/gin-contrib/cors"
	"github.com/gin-gonic/gin"
)

// CorsConfig returns a Gin middleware that configures CORS for the given frontend URL.
func CorsConfig(frontendURL string) gin.HandlerFunc {
	return cors.New(cors.Config{
		AllowOrigins:     []string{frontendURL},
		AllowMethods:     []string{"GET", "POST", "OPTIONS"},
		AllowHeaders:     []string{"Authorization", "Content-Type"},
		AllowCredentials: true,
	})
}

File: backend/src/middleware/cors.ts
import cors from "cors";

/**
 * フロントエンドオリジンのみ許可する CORS ミドルウェアを生成する。
 * @param frontendURL 許可するフロントエンドのオリジン。
 * @returns 設定済みの CORS ミドルウェア。
 */
export function corsConfig(frontendURL: string) {
  return cors({
    origin: frontendURL,
    methods: ["GET", "POST", "PUT", "OPTIONS"],
    allowedHeaders: ["Authorization", "Content-Type"],
    credentials: true,
  });
}
